src/m365/todo/commands/list/list-set.ts

```diff
@@ -26,37 +27,68 @@ class TodoListSetCommand extends GraphCommand {
     return 'Updates a Microsoft To Do task list';
   }
 
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.id = typeof args.options.id !== 'undefined';
+    telemetryProps.name = typeof args.options.name !== 'undefined';
+    return telemetryProps;
+  }
+
   public commandAction(cmd: CommandInstance, args: CommandArgs, cb: (err?: any) => void): void {
     const body: any = {
       displayName: args.options.newName
     };
 
-    const requestOptions: any = {
-      url: `${this.resource}/beta/me/todo/lists/${args.options.id}`,
-      headers: {
-        accept: 'application/json;odata.metadata=none',
-        'content-type': 'application/json'
-      },
-      body,
-      json: true
+    const getListId = () => {
+      if (args.options.name) {
+        const requestOptions: any = {
+          url: `${this.resource}/beta/me/todo/lists?$filter=displayName eq '${escape(args.options.name)}'`,
+          headers: {
+            accept: "application/json;odata.metadata=none"
+          },
+          json: true
+        };
+        return request
+          .get(requestOptions)
+          .then((response: any) => response.value && response.value.length === 1 ? response.value[0].id : null);
+      }
+
+      return Promise.resolve(args.options.id);
     };
 
-    request
-      .patch(requestOptions)
-      .then((): void => {
-        if (this.verbose) {
-          cmd.log(vorpal.chalk.green('DONE'));
-        }
+    getListId().then(listId => {
+      if (!listId) {
+        return Promise.reject(`The list ${args.options.name} cannot be found`);
+      }
+      const requestOptions: any = {
+        url: `${this.resource}/beta/me/todo/lists/${listId}`,
+        headers: {
+          accept: 'application/json;odata.metadata=none',
+          'content-type': 'application/json'
+        },
+        body,
+        json: true
+      };
+
+      return request.patch(requestOptions)
+    }).then((): void => {
+      if (this.verbose) {
+        cmd.log(vorpal.chalk.green('DONE'));
+      }
 
-        cb();
-      }, (err: any) => this.handleRejectedODataJsonPromise(err, cmd, cb));
+      cb();
+    }, (err: any) => this.handleRejectedODataJsonPromise(err, cmd, cb));
   }
 
   public options(): CommandOption[] {
     const options: CommandOption[] = [
       {
         option: '-i, --id <id>',
```

Comment:
Since id and name became optional, they should be designated as such using square brackets, eg. `[id]`